src/overlays/actors/ovl_En_Ba/z_en_ba.c

```diff
@@ -66,35 +76,450 @@ static ColliderJntSphInit D_809B80D4 = {
+    }
+    this->unk2FC = this->actor.world.pos;
+    if (globalCtx->gameplayFrames % 16 == 0) {
+        this->unk308.z += Rand_CenteredFloat(180.0f);
+        this->unk314 += Rand_CenteredFloat(180.0f);
+        this->unk308.x = Math_SinF(this->unk308.z) * 80.0f;
+        this->unk308.y = Math_CosF(this->unk314) * 80.0f;
+    }
+    this->unk2FC.y -= 448.0f;
+    this->unk2FC.x += this->unk308.x;
+    this->unk2FC.z += this->unk308.y;
+    func_80033AEC(&this->unk2FC, &this->unk158[13], 1.0f, this->actor.speedXZ, 0.0f, 0.0f);
+    for (i = 12; i >= 0; i--) {
+        func_80035844(&this->unk158[i + 1], &this->unk158[i], &sp5C, 0);
+        Matrix_Translate(this->unk158[i + 1].x, this->unk158[i + 1].y, this->unk158[i + 1].z, MTXMODE_NEW);
+        Matrix_RotateRPY(sp5C.x, sp5C.y, 0, MTXMODE_APPLY);
+        Matrix_MultVec3f(&D_809B8080, &this->unk158[i]);
+    }
+    func_80035844(&this->unk158[0], &this->unk2FC, &sp5C, 0);
+    Matrix_Translate(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z, MTXMODE_NEW);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->unk2A8[0].y, 3, this->unk31C, 182);
+    Math_SmoothStepToS(&this->actor.shape.rot.x, this->unk2A8[0].x, 3, this->unk31C, 182);
+    Matrix_RotateRPY(this->actor.shape.rot.x - 0x8000, this->actor.shape.rot.y, 0, MTXMODE_APPLY);
+    Matrix_MultVec3f(&D_809B8080, &this->unk158[0]);
+    this->unk2A8[13].y = sp5C.y;
+    this->unk2A8[13].x = sp5C.x + 0x8000;
+
+    for (i = 0; i < 13; i++) {
+        Matrix_Translate(this->unk158[i].x, this->unk158[i].y, this->unk158[i].z, MTXMODE_NEW);
+        Math_SmoothStepToS(&this->unk2A8[i].y, this->unk2A8[i + 1].y, 3, this->unk31C, 182);
+        Math_SmoothStepToS(&this->unk2A8[i].x, this->unk2A8[i + 1].x, 3, this->unk31C, 182);
+        Matrix_RotateRPY(this->unk2A8[i].x - 0x8000, this->unk2A8[i].y, 0, MTXMODE_APPLY);
+        Matrix_MultVec3f(&D_809B8080, &this->unk158[i + 1]);
+    }
+    this->unk2A8[13].x = this->unk2A8[12].x;
+    this->unk2A8[13].y = this->unk2A8[12].y;
+    if (!(player->stateFlags1 & 0x4000000) && (this->actor.xzDistToPlayer <= 175.0f) &&
+        (this->actor.world.pos.y == this->actor.home.pos.y + 100.0f)) {
+        EnBa_SetupSwingAtPlayer(this);
+    }
+}
+
+void EnBa_SetupFallAsBlob(EnBa* this) {
+    this->unk14C = 0;
+    this->actor.speedXZ = Rand_CenteredFloat(8.0f);
+    this->actor.world.rot.y = Rand_CenteredFloat(65535.0f);
+    this->unk318 = 20;
+    this->actor.gravity = -2.0f;
+    EnBa_SetupAction(this, EnBa_FallAsBlob);
+}
+
+/**
+ * Action function of the pink fleshy blobs that spawn and fall to the floor when a tentacle dies
+ */
+void EnBa_FallAsBlob(EnBa* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 1) {
+        this->actor.scale.y -= 0.001f;
+        this->actor.scale.x += 0.0005f;
+        this->actor.scale.z += 0.0005f;
+        this->unk318--;
+        if (this->unk318 == 0) {
+            Actor_Kill(&this->actor);
+            return;
```

Comment:
This return is redundant. Do we keep it anyways because of actor_kill?

Reply:
in this case the else is needed anyway so it cant be phrased as an early return. can remove